cpu/stm32/kconfigs/Kconfig.clk

```
@@ -28,31 +28,53 @@ config USE_CLOCK_HSI
 
 endchoice
 
-if CPU_FAM_G0 || CPU_FAM_G4
+if CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
+choice
+bool "Source clock for PLL" if USE_CLOCK_PLL
+default CLOCK_PLL_SRC_MSI if !BOARD_HAS_HSE
+default CLOCK_PLL_SRC_HSE if BOARD_HAS_HSE
+
+config CLOCK_PLL_SRC_MSI
+    bool "Use MSI source clock"
+
+config CLOCK_PLL_SRC_HSE
+    bool "Use HSE source clock"
+    depends on BOARD_HAS_HSE
+
+config CLOCK_PLL_SRC_HSI
+    bool "Use HSI16 source clock"
+endchoice
+
+endif  # CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
+
+if CPU_FAM_G0 || CPU_FAM_G4 || CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 config CLOCK_PLL_M
     int "M: PLLIN division factor" if USE_CLOCK_PLL
     default 1 if CPU_FAM_G0
-    default 4 if CPU_FAM_G4
+    default 4 if CPU_FAM_G4 && !BOARD_HAS_HSE
     default 6 if CPU_FAM_G4 && BOARD_HAS_HSE
```

Defaults are evaluated from top to bottom, this should work instead right?
```suggestion
    default 6 if CPU_FAM_G4 && BOARD_HAS_HSE
    default 4 if CPU_FAM_G4
```

---

Ah thanks! I had a hard time making the options work depending on the families :)

This is fixed now. I also adapted the default value PLLN value for G4. The value in master only allows to reach 80MHz and is not the same as the one used by the board (which allows to reach 170MHz).